popup.js

```diff
@@ -0,0 +1,87 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const runScriptButton = document.getElementById('runScript');
+  const postList = document.getElementById('postList');
+  const noPosts = document.getElementById('noPosts');
+  if (runScriptButton) {
+    runScriptButton.addEventListener('click', async () => {
+      postList.innerHTML = '';
+      noPosts.style.display = 'none';
+      const [tab] = await chrome.tabs.query({ active: true, currentWindow: true });
+
+      if (tab.id) {

```

Comment:
this if statement seems to be so complex.... try to reduce complexity for better maintainability and readability...

simple idea to reduce complexity is it extract this in separate function, and leverage early return.